shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,110 @@
+package ro.msg.learning.shop.service;
+
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.exception.BadRequestException;
+import ro.msg.learning.shop.exception.NotFoundException;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.mapper.StockMapper;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static ro.msg.learning.shop.message.Messages.*;
+
+@Service
+@AllArgsConstructor
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    private final ProductCategoryRepository productCategoryRepository;
+
+    private final StockRepository stockRepository;
+
+    private final LocationRepository locationRepository;
+
+
+    private final ProductMapper productMapper;
+
+    private final StockMapper stockMapper;
+
+
+    public ProductDto createProduct(ProductDto productDto) throws BadRequestException {
+
+        if (productCategoryRepository.findById(productDto.getCategory()).isEmpty()) {
+            throw new BadRequestException(BAD_PRODUCT_CATEGORY);
+        }
+        ProductCategory productCategory = productCategoryRepository.findById(productDto.getCategory()).get();
+        productDto.setCategory(productCategory.getProductCategoryId());
+        productRepository.save(productMapper.toEntity(productDto));
+        return productDto;
+    }
+
+
+    public String deleteProduct(UUID productId) throws NotFoundException {
```

Comment:
For example here in this method you use findById() from product repository.
You don't need 3 calls to it. You can define it at the beginning and then use it:
`Optional<Product> product = productRepository.findById(productId);`
Then you can use it everywhere:
- product.isEmpty()
- product.get()
- product.isPresent()